arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/OptionTest.kt

```diff
@@ -143,10 +178,141 @@ class OptionTest : UnitSpec() {
       none.toList() shouldBe listOf()
     }
 
-    "firstOption" {
-      val l = listOf(1, 2, 3, 4, 5, 6)
-      l.firstOrNone() shouldBe Some(1)
-      l.firstOrNone { it > 2 } shouldBe Some(3)
+    "Iterable.firstOrNone" {
+      val iterable = object : Iterable<Int> {
+        private val list = listOf(1, 2, 3, 4, 5, 6)
+        override fun iterator(): Iterator<Int> = list.iterator()
+      }
+      iterable.firstOrNone() shouldBe Some(1)
+      iterable.firstOrNone { it > 2 } shouldBe Some(3)
+      iterable.firstOrNone { it > 7 } shouldBe None
+
+      val emptyIterable = object : Iterable<Int> {
+        private val emptyList = emptyList<Int>()
+        override fun iterator(): Iterator<Int> = emptyList.iterator()
+      }
```

Comment:
```suggestion
      val emptyIterable = Iterable { iterator { yieldAll(listOf(1, 2, 3, 4, 5, 6)) } }
```

These and other interators can be defined like this :point\_up\_2:   
  
Alternatively, we can also define a file function like this:

```kotlin
fun <T> nonCollectionOf(vararg items: T): Iterable<T> = Iterable { iterator { yieldAll(items.toList()) } }
```

Reply:
Definitely going to use this, really helpful :+1: